tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/temporal/TableauEncoder.scala

```diff
@@ -33,6 +34,21 @@ class TableauEncoder(
 
   private def inBoolSet(element: TBuilderInstruction): TBuilderInstruction = builder.in(element, builder.booleanSet())
 
+  /* replaces all occurences of foo := bar with foo = bar */
```

Comment:
could you add a comment on why it is sound?

Reply:
You mean, why this is okay to do?
This is inside a temporal property, not in an actual action. Leaving these as assignments leads the transition finder to see an assignment on the right-hand side of another assignment.
E.g. consider this
```
CoolAction == x := 2

TemporalProperty == <>[CoolAction]_x
```

Due to the details of the encoding, a variable will be produced that gets assigned like this:
```
_temporal__CoolAction := (x := 2)
```

Then the assignment (x := 2) should be written as an equality, and that's what this function does

Reply:
I mean, usually you would have `x' := 2`, that is `x` with a prime. Would you keep the prime?

Reply:
see https://github.com/informalsystems/apalache/blob/6d5a5280aa66b7f3584188ea810f40c6447411f7/tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/temporal/TableauEncoder.scala#L388

Reply:
Clarified this a bit more. (TLDR: everything is kept the same, except := is replaced by =)